monadic-html/src/main/scala/scala/xml/xml.scala

```diff
@@ -76,27 +76,49 @@ final case class PrefixedAttribute(
     this(pre, key, Text(value), next)
 }
 
-final case class UnprefixedAttribute(
+final case class UnprefixedAttribute[T] private (
   key: String,
   value: Node,
   next: MetaData
 ) extends MetaData {
-  def this(key: String, value: String, next: MetaData) =
-    this(key, Text(value), next)
+  def this(key: String, e: T, next: MetaData)(implicit ev: XmlSerializable[T]) =
+    this(key, ev.toNode(e), next)
 }
 
-// Internal structure used by scalac to create literals -----------------------
+@scala.annotation.implicitNotFound(msg =
+  """Cannot embed value of type '${T}' into an xml literal. Allowed types are:
+Base types:
+  - String (tip: you can use ${T}.toString)
+  - Int
+  - xml.Node
+  - List[xml.Node], Seq[xml.Node] and other Iterable containers.
+Event handlers:
+  - () => Unit
+  - T => Unit, where T can be any type.
+Monadic-html:
+  - mhtml.Rx[T] or mhtml.Var[T], where T is a type that can be embedded into an xml literal. For example, a String or Seq[xml.Node].
+""")
+sealed trait XmlSerializable[T] { def toNode(e: T): Node }
```

Comment:
I think it could be nice to keep `scala-xml-light` and `mhtml` separated if possible, it would at least look less hacky from outside to be able to say "hey looks this is a light, scalac compatible alternative to `scala-xml`" and not something deeply embedded into the lib...